client/src/main/java/io/pravega/client/BatchClientFactory.java

```diff
@@ -103,4 +104,14 @@ static BatchClientFactory withScope(String scope, ClientConfig config) {
      * @return A list of segment range in between a start and end stream cut.
      */
     List<SegmentRange> getSegmentRangeBetweenStreamCuts(final StreamCut startStreamCut, final StreamCut endStreamCut);
+
+    /**
+     * Provides nearest streamcut in future depending on the distance and current streamcut.
```

Comment:
It will be great if we can provide java doc on how stream cut been calculated per segment for better clarity.

Reply:
Done.